src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -108,9 +109,16 @@ public InitCodeView generateMockResponseObjectInitCode(
   }
 
   private InitCodeView buildInitCodeView(MethodTransformerContext context, InitCode initCode) {
+    ModelTypeTable sampleTypeTable = generateInitCodeTypeTable(context);
```

Comment:
MethodTransformerContext already has a ModelTypeTable in it. We should just use that. If sample code always needs a separate type table (I think this may be the case), we should make the init code transformer create a new MethodTransformerContext with a newly-created type table at each entry point. This could be done with a method in MethodTransformerContext to create a copy of itself in that fashion, maybe called "cloneWithEmptyTypeTable" or something like that.